chainerio/__init__.py

```diff
@@ -68,60 +87,194 @@ def open(file_path: str, mode: str = 'rb',
+    the name of the handler.
+
+    Example::
+
+        # Case 1
+        # set_root by the name of uri
+        chainerio.set_root("posix")
+        # open a file on posix filesystem with path "some/file"
+        chainerio.open("some/file")
+        # override with a full uri
+        chainerio.open("hdfs:///some/file/on/hdfs")
+
+        # Case 2
+        # set_root by uri
+        chainerio.set_root("hdfs:///some/directory")
+        # open a file "on/hdfs" on hdfs under the " some/directory"
+        chainerio.open("on/hdfs")
+
+        # Case 3
+        handler = chainerio.create_handler('hdfs')
+        # set_root by handler
+        chainerio.set_root(handler)
+        # open a file "/some/file/on/hdfs" on hdfs
+        chainerio.open("/some/file/on/hdfs")
+
+        # Case 4
+        handler = chainerio.open_as_container('some.zip')
+        # set_root by handler
+        chainerio.set_root(handler)
+        # open a file "img.jpg" in "some.zip"
+        chainerio.open("img.jpg")
+
+    The root path refers to a directory that ChainerIO works on.
+    It is similar to current working directory, ``CWD`` in terms of the shell
+    environment.
+    The root path will only be set
+    when the ``uri_or_handler`` points to a directory.
+    Otherwise, it will be set to default,
+    which represents the default working directory as follow:
+
+    +---------+---------------------------+
+    |         |     Default Root Path     |
+    +=========+===========================+
+    | POSIX   | current working directory |
+    +---------+---------------------------+
+    | HDFS    |     /user/``USERNAME`` #1 |
+    +---------+---------------------------+
+    | zip     |      top directory        |
+    +---------+---------------------------+
+
+    #1 For the details about the ``USERNAME`` in HDFS, please refer to
+    `HDFS Document <https://hadoop.apache.org/docs/current/\
+            hadoop-project-dist/hadoop-hdfs/HdfsPermissionsGuide.html>`_
+
+
+    Args:
+        uri_or_handler (str or :class:~`chainerio.IO`): The ``uri_or_handler``
+            can accept the following three kinds of values:
+
+            1. the name of a handler (string): set the default handler
+                to the corresponding handler, and root path.
+                See :func:`create_handler` for supported name of handlers.
+
+            2. an uri of directory (string): set the context
+                to use the corresponding handler and set
+                the root path to the given directory.
+
+            3. a handler, which is an instance of
+                :class:`chainerio.IO`. Set the default handler
+                to the given handler, and root path to default.
+
+    """
     global _DEFAULT_CONTEXT
     default_context = _DEFAULT_CONTEXT
 
     default_context.set_root(uri_or_handler)
 
 
-def create_handler(uri: str) -> IO:
+def create_handler(handler_name: str) -> IO:
+    """Returns a handler according to the given ``handler_name``
```

Comment:
Aren't they scheme defined here https://github.com/chainer/chainerio/blob/master/docs/source/design.rst#uri-expression-of-file-paths technically?

Reply:
Yes, I think.